### DISCUSSION TITLE
編集中のMarkdown書式レコメンド機能について

### DISCUSSION THREAD

**Discussion by GleamDream** [increments/qiita-discussions#299] [要望・アイデア・不具合]
近年，単純な短いコードの需要が減りつつあって，色んな記事/質問で長いコードを見かけるのが増えました．

私の場合[^1]，長いコードは記事全体の内容の見通しが悪くなったり，
スクロール量が増えたりするため，
`details`タグを使って折り畳むようにしています．

そこで，Markdownの書き方をあまり知らない人でも折り畳めるように，
コードブロック内に記述したコード量が一定量を超えた場合，
`details`タグを使って折り畳んでもらうようにするレコメンド機能[^2]があると嬉しく思います．

折り畳みに限らず，
行頭の全角`・`を検出してListの書式をレコメンドしたり，
全角ギリシャ文字や四則演算記号を検出して数式モードをレコメンドしたりなど，
現在のテーブル入力補完の技術を応用できれば可能かと思いました．

[^1]: 人によっては「最初にコード全文，後に各部詳細説明」をされる方もいれば，「最初に各部の説明，最後にコード全文の掲載」をする人もいます．後者は問題ないと感じますが特に前者は記事が追いづらいと感じました．
[^2]: 何度も表示するのは編集作業に影響が出ると思いますので，レコメンド後に「もう表示しない」などのボタンがあれば嬉しいです．

**Comment by kskwtnk** [comment]
ご意見ありがとうございます！

記事の書き方そのものは色々なスタイルがあるため、一律にレコメンドすることは考えておりませんでした 🙏 

ただ、そもそも`details`による折り畳みを知らないですとか、Markdownの書式をあまり知らないなどで`・`によるリスト表記をしてしまうとか、そういった事象はありそうです。
アカウント作成時のオンボーディングやフォーマッターのような機能など、少し別な角度からアプローチさせていただければ、と思います 🙏

**Reply by GleamDream** [reply]
確かに，記事を書く個人のスタイルを尊重できている今のままの方が良さそうな気がしてきました．
オンボーディングやフォーマッタ機能など，楽しみにしております．
ご検討いただきありがとうございます．
